provider/internal/builtin/provider.go

```diff
@@ -51,14 +51,23 @@ var capabilities = []provider.Capability{
 }
 
 type builtinCondition struct {
-	Filecontent              string        `yaml:"filecontent"`
-	File                     string        `yaml:"file"`
-	XML                      xmlCondition  `yaml:"xml"`
-	JSON                     jsonCondition `yaml:"json"`
-	HasTags                  []string      `yaml:"hasTags"`
+	Filecontent              fileContentCondition `yaml:"filecontent"`
+	File                     fileCondition        `yaml:"file"`
+	XML                      xmlCondition         `yaml:"xml"`
+	JSON                     jsonCondition        `yaml:"json"`
+	HasTags                  []string             `yaml:"hasTags"`
 	provider.ProviderContext `yaml:",inline"`
 }
 
+type fileContentCondition struct {
+	FilePattern string `yaml:"filePattern"`
```

Comment:
naming on this one is a little awkward, probably ok though

Reply:
Want to brainstorm?